examples/s4u/async-waitany/s4u_async-waitany.cpp

```diff
@@ -0,0 +1,108 @@
+/* Copyright (c) 2010-2016. The SimGrid Team. All rights reserved.          */
+
+/* This program is free software; you can redistribute it and/or modify it
+ * under the terms of the license (GNU LGPL) which comes with this package. */
+
+#include "simgrid/s4u.hpp"
+#include "xbt/str.h"  
+#include <cstdlib>
+#include <iostream>
+#include <vector>
+
+XBT_LOG_NEW_DEFAULT_CATEGORY(msg_async_waitany, "Messages specific for this msg example");
+
+class sender {
+public:
+  explicit sender(std::vector<std::string> args)
+{
+  xbt_assert(args.size()== 5, "This function expects 5 parameters from the XML deployment file");
+  long number_of_tasks = xbt_str_parse_int(args[0].c_str(), "Invalid amount of tasks: %s");
+  double task_comp_size = xbt_str_parse_double(args[1].c_str(), "Invalid computational size: %s");
+  double task_comm_size = xbt_str_parse_double(args[2].c_str(), "Invalid communication size: %s");
+  long receivers_count = xbt_str_parse_int(args[3].c_str(), "Invalid amount of receivers: %s");
+  int diff_com = xbt_str_parse_int(args[4].c_str(), "Invalid value for diff_comm: %s");
+
+  std::vector<simgrid::s4u::CommPtr> comms;
+  simgrid::s4u::CommPtr comm;
+  simgrid::s4u::MailboxPtr mbox = simgrid::s4u::Mailbox::byName("receiver_mailbox");
+  /* First pack the communications in the dynar */
+  for (int i = 0; i < number_of_tasks; i++) {
+    double coef = (diff_com == 0) ? 1 : (i + 1);
+
+    char mailbox[80];
+    char taskname[80];
+    snprintf(mailbox,79, "receiver-%ld", (i % receivers_count));
+    snprintf(taskname,79, "Task_%d", i);
+
+    comm = mbox->put_async((void*)taskname, 42.0);
+    comms.push_back(comm);
+    XBT_INFO("Send to receiver-%ld %s comm_size %f", i % receivers_count, taskname, task_comm_size / coef);
+  }
+
+  /* Here we are waiting for the completion of all communications */
+  while (!comms.empty()) {
+    comms.pop_back();
```

Comment:
You are missing a call to comm->wait() here

Reply:
I think that we don't have to put wait over here.